Diplom,develop_3,UI tests

railvaliullin opened the pull request.

railvaliullin committed:
Diplom,develop_3,UI tests

egorvas commented on src/test/java/tests/login/LogoutTest.java:
```diff
@@ -0,0 +1,62 @@
+        new WebDriverWait(driver, Duration.ofSeconds(5))
+                .until(ExpectedConditions.urlContains("/login"));
+
+        driver.get("https://stellarburgers.nomoreparties.site/login");
```
**Нужно исправить:** очень плохая практика хардкодить URL. Ладно бы еще ты это в одном месте сделал, но если завтра у сайта поменяется URL или нужно будет прогнать тесты на тестовом стенде, то придется менять в куче мест.
Лучше всего задавать URL параметром запуска.

egorvas commented on src/test/java/tests/navigation/NavigationFromAccountTest.java:
```diff
@@ -0,0 +1,71 @@
+    private LoginPage loginPage;
+    private AccountPage accountPage;
+
+    private final String EMAIL = "rail_test@gmail.com";
```
**Можно лучше:** Вместо того чтобы самому придумывать тестовые строки я рекомендую использовать модули для генерации данных, например java faker: https://github.com/DiUS/java-faker
С его помощью ты сможешь убрать большую часть хардкода из твоих тестов.

egorvas commented on src/test/java/tests/navigation/NavigationFromAccountTest.java:
```diff
@@ -0,0 +1,71 @@
+        new WebDriverWait(driver, Duration.ofSeconds(5))
+                .until(ExpectedConditions.urlContains("/"));
+        Assert.assertTrue("Должны быть на главной странице",
+                driver.getCurrentUrl().endsWith("/"));
```
**Нужно исправить:** проверка на то что URL кончается "/" честно говоря, ну крайне абсурдная. А если тебя вообще на другой сайт перекинет где на "/" будет кончаться?

railvaliullin committed:
Diplom,develop_3,UI tests. Исправления:
Вынес URL в отдельный класс.Для генерации данных добавил java faker.Поправил проверку URL
